test/python/circuit/test_bit.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=missing-function-docstring
+
+"""Test library of quantum circuits."""
+
+from unittest import mock
+
+from qiskit.test import QiskitTestCase
+from qiskit.circuit import bit
+from qiskit.circuit import quantumregister
+from qiskit.circuit import classicalregister
+
+
+class TestBitClass(QiskitTestCase):
+    """Test library of boolean logic quantum circuits."""
+
+    def test_bit_hash_update_reg(self):
+        orig_reg = mock.MagicMock()
+        orig_reg.size = 3
```

Comment:
Is the size set on the mock used? Maybe it would be clearer to explicitly mock its `__hash__`.

Reply:
It is because it's part of the register `__hash__` which is the first component of a Bit's `__hash__`. I'm not sure if mocking `Register.__hash__` is clearer, we're pretty much in the weeds on this already, especially since these classes are basically static. I also worry about sticky state with that because we'll need to change the return_value for the mock during the test between 2 consecutive calls.

Reply:
sorry, this was actually not correct, I remembered incorrectly. It's because as part of a Bit class's `__init__` method it compares the index it's being initialized with against it's register size attribute to see if it exceeds the bounds. With the mock class this causes a failure because it can't compare the mock type with an integer.